stdlib/toolchain/CMakeLists.txt

```diff
@@ -48,8 +48,17 @@ list(APPEND CXX_COMPILE_FLAGS "-DSWIFT_COMPATIBILITY_LIBRARY=1")
 
 if(SWIFT_STDLIB_SUPPORT_BACK_DEPLOYMENT)
   add_subdirectory(legacy_layouts)
-  add_subdirectory(Compatibility50)
-  add_subdirectory(Compatibility51)
-  add_subdirectory(CompatibilityDynamicReplacements)
-  add_subdirectory(CompatibilityConcurrency)
+  foreach(compatibility_name IN LISTS COMPATIBILITY_NAMES)
```

Comment:
This section is way harder to read now.

I agree with @compnerd here. Something like the following is a bit more repetitive, though repetitions are kept close in the source, but it's way easier to read and know what's going on.
```
add_subdirectory(Compatibility50)
add_subdirectory(Compatibility51)
add_subdirectory(CompatibilityDynamicReplacements)
add_subdirectory(CompatibilityConcurrency)

add_library(HostCompatibilityLibs INTERFACE) 
target_link_libraries(HostCompatibilityLibs INTERFACE
  swiftCompatibilityConcurrency${vsuffix}
  swiftCompatibilityDynamicReplacements${vsuffix}
  swiftCompatibility50${vsuffix}
  swiftCompatibility51${vsuffix})
```
 
As long as the repetitions are kept close together in the code, it's manageable. The issue is having to remember to keep multiple files in sync with each other. Then those who need the compat libs can link against the `HostCompatibilityLibrary` and it's easy.

Reply:
I admit in devising my initial change I thought I had to have a global variable (like in a general programming language) to reuse this in `SwiftCompilerSources/CMakeLists.txt` -- but in fact we can have that neatly defined in the proper place, and CMake will be able to find the target at generation time.

This has been reworked in 8b7d9152b7d -- let me know if that looks better.